containers/unit_tests/TestVector.hpp

```diff
@@ -52,6 +52,70 @@ namespace Test {
 
 namespace Impl {
 
+template <typename Scalar, class Device>
+struct test_vector_insert {
+  typedef Scalar scalar_type;
+  typedef Device execution_space;
+
+  template <typename Vector>
+  void run_test(Vector& a) {
+    int n = a.size();
+
+    auto it = a.begin();
+    it += 15;
+    ASSERT_EQ(*it, scalar_type(1));
+
+    a.insert(it, scalar_type(3));
+    ASSERT_EQ(a.size(), n + 1);
+
+    it = a.begin();
+    it += 17;
+    a.insert(it, 5, scalar_type(5));
+    ASSERT_EQ(a.size(), n + 6);
+
+    Vector b(7, 9);
+    it = a.begin();
+    it += 27;
+    a.insert(it, b.begin(), b.end());
+    ASSERT_EQ(a.size(), n + 13);
```

Comment:
You need a test that inserts at least `a.size() * a._extra_storage` items in order to test iterator invalidation.

Reply:
Yeah I should add that.